Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -883,23 +960,43 @@ private OrderResponse HandleCancelOrderRequest(CancelOrderRequest request)
 
         private void HandleOrderEvent(OrderEvent fill)
         {
-            // update the order status
-            var order = GetOrderByIdInternal(fill.OrderId);
-            if (order == null)
+            Order order;
+            OrderTicket ticket;
+            if (fill.Status.IsClosed())
             {
-                Log.Error("BrokerageTransactionHandler.HandleOrderEvent(): Unable to locate Order with id " + fill.OrderId);
-                return;
+                if(_openOrders.TryRemove(fill.OrderId, out order)) {
+                    _closedOrders.TryAdd(order.Id, order);
```

Comment:
What are your thoughts on using the indexer syntax here: `_closedOrders[order.Id] = order`

My thinking is that even if it exists we would want to overwrite what's there, granted, given the logic in this class this case should never happen. Likewise for order tickets below.

Reply:
Your right, better be in the safe side. Done